Remove zero start value and explain alternative. (#3226)
modelica/ModelicaSpecification: chapters/equations.tex

\lstinline!start!-values of variables having \lstinline!fixed = false! can be used as initial guesses, in case iterative solvers are used in the initialization phase.

\begin{nonnormative}

> In case of iterative solver failure, it is recommended to specially report those variables for which the solver needs an initial guess, but which only have the default value as defined in \cref{predefined-types-and-classes}, since the lack of appropriate initial guesses is a likely cause of the solver failure.

Reviewer: It's the new fallback value we're talking about here, right?
```suggestion
In case of iterative solver failure, it is recommended to specially report those variables for which the solver needs an initial guess, but where the fallback value (see \cref{predefined-types-and-classes}) has been applied, since the lack of appropriate initial guesses is a likely cause of the solver failure.
```

\end{nonnormative}

> If a parameter has a value for the \lstinline!start!-attribute, does not have \lstinline!fixed = false!, and neither has a binding equation nor is part of a record having a binding equation, the value for the \lstinline!start!-attribute can be used to add a parameter binding equation assigning the parameter to that \lstinline!start! value.

Reviewer: Maybe we could simplify our language further?
```suggestion
If a parameter has a \lstinline!start!-attribute, does not have \lstinline!fixed = false!, and neither has a binding equation nor is part of a record having a binding equation, the \lstinline!start!-attribute can be used to add a parameter binding equation assigning the parameter to that value.
```

Reviewer: Yes, and it is in the light of this that I think we should take the chance to find an easier way to express ourselves; everyone knows that every `Real` variable has all these attributes, so it feels safe to say _has an X-attribute_ when meaning that the attribute has a value.  I think this would be one of the real benefits of this PR, allowing us to take a long step away from the complicated _has a modifier for the X-attribute_.

Reviewer: By the way, how to word a thing like this seems like a perfect question to ask the language group.

Author: Ok, but preferably separate from this PR; as it will likely impact many other texts.

Reviewer: OK, I'll create a separate issue.  Resolving.

In this case a diagnostic message is recommended in a simulation model.

\begin{nonnormative}

> This is used in libraries to give acceptable defaults so that users can quickly combine models and simulate without setting parameters; but still easily find the parameters that need to be set.

Reviewer: This still sounds like a contradiction between both having OK defaults, and at the same time these defaults not being sufficient.

To mark more clearly that these defaults are not defaults in the full sense of providing a value for the parameter, I suggest minor rewording:
```suggestion
This is used in libraries to give reudimentary defaults so that users can quickly combine models and simulate without setting parameters; but still easily find the parameters that should be set properly.
```

> This is used in libraries to give appropriate defaults so that users can quickly combine models and simulate without setting parameters; but still easily find the parameters that need to be set.

Reviewer: If the defaults would have been "appropriate", it would have been more natural to provide them as value modifiers instead.  Can we find a formulation that doesn't suggest that this is good modeling style?
```suggestion
This is used in libraries to give dummy defaults so that users can quickly combine models and simulate without setting parameters; but still easily find the parameters that need to be set.
```

Author: Dummy is too far off for me, as it indicates it isn't really a default. After using a thesaurus I have the alternatives:

- Acceptable
- Permissible
- Tolerable
- Adequate

I believe the first two are acceptable.

Reviewer: Maybe what they are is simply _defaluts_, but I see a risk of causing confusion as setting a value for the parameter is probably also considered giving a default:
```
  parameter SI.Resistance R = 1; /* Here, the 1 is also a default in one sense */
```

Would it be better to call it a _fallback value_?  (Also, these parameters obviously don't need to be set, as there is a fallback.)
```suggestion
This is used in libraries to provide a fallback value so that users can quickly combine models and simulate without setting parameters; but still easily find the parameters that should be set properly.
```

Author: I changed to _acceptable_.
